xarray/tests/test_dataarray.py

```diff
@@ -3153,6 +3153,22 @@ def test_full_like(self):
         assert expect.dtype == bool
         assert_identical(expect, actual)
 
+    def test_label_like(self):
+        # For more thorough tests, see test_variable.py
+        orig = DataArray(np.random.random(size=(2, 2)),
+                         dims=('x', 'y'),
+                         attrs={'attr1': 'value1'},
+                         coords={'x': [4, 3]},
+                         name='helloworld')
+        new_data = np.arange(4).reshape(2, 2)
+        new_var = label_like(new_data, orig)
+        for i in ['dims', 'attrs', 'name']:
+            assert getattr(new_var, i) == getattr(orig, i)
+        for coord in new_var.coords:
+            assert coord in orig.coords
+        with raises_regex(AssertionError, 'Arrays are not equal'):
+            assert_array_equal(new_var, orig)
```

Comment:
This could be a good use case for `xarray.testing.assert_identical()`. You would just need to prepare an object manually to compare identical to (e.g., by copying `orig` and replacing its `data`).

Reply:
yeah that is a lot better.